### DISCUSSION TITLE
Change Gltf transform

### DISCUSSION THREAD

**Discussion by salah-walid** [iTwin/itwinjs-core#3605] [Q&A] [answered]
Hi,

Following [this example](https://www.itwinjs.org/sandboxes/NathanElrod/glTF%20Decorator) I created a scene where i spawn multiple GLTF models.

Now i want a way to update the Gltf transform with a slider or an input.
Is there a way to access the transform in the GraphicBranch object ?

**Comment by pmconne** [comment, marked as answer]
You can't change the transform after creating the branch. But you can store the glTF graphic instead of the branch, and create the branch in your `decorate` method to control its transform.
[This sample](https://www.itwinjs.org/sandbox/pmconne/Transform%20glTF%20decoration) illustrates that approach.

**Reply by salah-walid** [reply]
Thanks, it worked, but after implementing this collision detection isn't working, neither for tooltip nor for the onDecorationButtonEvent event in the decorator

**Reply by pmconne** [reply]
Thanks for reporting that - I'll have to investigate.
